### PR TITLE
Document Azure CLI setup for AKS with Entra Workload ID

### PR DESCRIPTION
**Pull request by greedy52**
Related PR:
- #40142 

Changes:
- Refactored existing Azure CLI access guide to have an intro + how it works
- Added guide for Azure CLI on AKS workload ID deployment

### CODE REVIEW

File: docs/pages/application-access/cloud-apis/azure.mdx [hunk 1]
```diff
@@ -30,6 +26,15 @@ prevent unauthorized access to your organization's Azure identities.
 
 - An Azure VM where you will run the Teleport Application Service. The Azure VM
   must be running a Linux distribution.
+  
+  <Admonition type="note" title="Azure Kubernetes Service (AKS)">
+  Note that this guide is also applicable to Azure Kubernetes Service (AKS)
+  deployment with Microsoft Entra pod-managed identities enabled. However, the
+  pod-managed identities feature will deprecate in September 2024.
```
Comment (Tener) [inline docs/pages/application-access/cloud-apis/azure.mdx:33]: Nit: I don't remember seeing `deprecate` used in this fashion... perhaps we can rewrite this way?

```suggestion
  pod-managed identities feature will be deprecated in September 2024.
```

File: docs/pages/includes/application-access/azure-tsh.mdx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+```
+
+This command validates the value of the `--azure-identity` flag against the ones
+the user is authorized to assume. The value of the flag can either be the full
+URI of the identity (e.g., the URI you copied earlier in this guide) or the name
+of the identity, e.g., `teleport-azure`. 
+
+A user can omit the `--azure-identity` flag if they are only authorized to
+access a single Azure identity, but otherwise an empty `--azure-identity` will
+result in an error.
```
Comment (Tener) [inline docs/pages/includes/application-access/azure-tsh.mdx:33]: Technically omitting the `--azure-identity` flag will be seen as an empty value by `tsh`, but I'd rephrase this as the flag being "missing" since this is the most likely cause of an empty value.

File: docs/pages/includes/application-access/azure-tsh.mdx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+```text
+Started Azure proxy on http://127.0.0.1:54330.
+To avoid port randomization, you can choose the listening port using the --port flag.
+
+Use the following credentials and HTTPS proxy setting to connect to the proxy:
+
+  export AZURE_CONFIG_DIR=/Users/myuser/.tsh/azure/my.teleport.cluster/azure
+  export HTTPS_PROXY=http://127.0.0.1:54330
+  export HTTP_PROXY=http://127.0.0.1:54330
+  export MSI_ENDPOINT=https://azure-msi.teleport.dev/eedfd5b55257c0aaa58f
```
Comment (Tener) [inline docs/pages/includes/application-access/azure-tsh.mdx:115]: Make it explicit hex?

```suggestion
  export MSI_ENDPOINT=https://azure-msi.teleport.dev/123456789abcdef01234
```

**Review by Tener — APPROVED** [review]
Kudos for the clarity of the doc! Some minor points, but looks very nice overall.

File: docs/pages/application-access/cloud-apis/azure-aks-workload-id.mdx [hunk 4]
```diff
@@ -0,0 +1,239 @@
+---
+title: "Protect Azure CLIs with Teleport Application Access on AKS"
+description: How to enable secure access to Azure CLIs on Azure Kubernetes Service with Workload Identity.
+---
+
+(!docs/pages/includes/application-access/azure-intro.mdx!)
```
Comment (ptgott) [inline docs/pages/application-access/cloud-apis/azure-aks-workload-id.mdx:6]: The intro paragraph could use a smoother transition to the numbered list. We go from a description of the ability to protect CLI tools with Teleport to a list of objectives in the guide:

image

File: docs/pages/application-access/cloud-apis/azure-aks-workload-id.mdx [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+  for how to install the `az` CLI on your operating system.
+
+- `kubectl` and `helm` for AKS deployments.
+
+- (!docs/pages/includes/tctl.mdx!)
+
+
+## Step 1/6. Create an Azure managed identity for Teleport Application Service
+
+Teleport Application Service requires a managed identity that can retrieve the
```
Comment (ptgott) [inline docs/pages/application-access/cloud-apis/azure-aks-workload-id.mdx:41]:
```suggestion
The Teleport Application Service requires a managed identity that can retrieve the
```

File: docs/pages/application-access/cloud-apis/azure-aks-workload-id.mdx [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+client IDs of the managed identities for user access. This managed identity
+will be assigned as the default identity for the Kubernetes service account.
+
+Login to your Azure admin account with `az login` command if you haven't
```
Comment (ptgott) [inline docs/pages/application-access/cloud-apis/azure-aks-workload-id.mdx:45]:
```suggestion
Log in to your Azure admin account with `az login` command if you haven't
```

"Log in" is the verb phrase. "Login" is the noun.

File: docs/pages/application-access/cloud-apis/azure-aks-workload-id.mdx [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+Before using `kubectl`, make sure your local Kubernetes config is updated to
+access your AKS cluster:
+```code
+$ az aks get-credentials -n "${AKS_CLUSTER_NAME}" -g "${RESOURCE_GROUP}"
+```
+
+Create a Kubernetes service account and annotate it with the client ID of
+the managed identity created in the previous step:
+```code
+$ cat <<EOF | kubectl apply -f -
```
Comment (ptgott) [inline docs/pages/application-access/cloud-apis/azure-aks-workload-id.mdx:96]: It might make sense to separate the steps of creating a manifest and applying it so the user can review the YAML file. `Var`s can be a little tricky to use for readers who aren't familiar with them, so it'd be helpful to give the reader a chance to make sure spacing is correct etc..

**Review by ptgott — APPROVED** [review]
